### ISSUE TITLE
make load and error an event?

### ISSUE THREAD

**Issue by mesqueeb** [quasarframework/app-extension-qpdfviewer#7]
Hi! Thanks for this!!

I was just wondering instead of
```html
<q-pdfviewer
      :value="true"
      :src="src"
      :error="onError"
      :load="onLoad"
      content-class="fit _container"
      inner-content-class="fit _container"
    />
```
isn't it more natural to have `error` and `load` be events like so:
```html
    <q-pdfviewer
      :value="true"
      :src="src"
      @error="onError"
      @load="onLoad"
      content-class="fit _container"
      inner-content-class="fit _container"
    />
```

**Comment by hawkeye64** [comment]
Yes. That being said, I will be investigating PDF.js so things may change. What this will give us is:
1) Ability to display PDF on browsers (and version) that doen't support embedded PDF
2) Ability to display PDF in am Electron app

**Comment by rrigaud** [comment]
> Yes. That being said, I will be investigating PDF.js so things may change. What this will give us is:
> 
> 2. Ability to display PDF in am Electron app

Does that mean today, this extension doesn't work with Electron App ?
Because I fail using it...
`Cannot GET /Users/raph/Library/Application%20Support/DroopyTempCut/file_1.pdf`

I can't preview any PDF, with local or distant URL... It seems that the URL isn't good, the console said :

`GET http://localhost:8080/Users/raph/Library/Application%20Support/DroopyTempCut/file_1.pdf 404 (Not Found)`

Sorry for my english, but I was so excited by using this Quasar Extension, it's a really important to me to be able to preview a PDF file inside my App...

Thanks a lot for your work.

**Comment by yelena870513** [comment]
> > Yes. That being said, I will be investigating PDF.js so things may change. What this will give us is:
> > 
> > 1. Ability to display PDF in am Electron app
> 
> Does that mean today, this extension doesn't work with Electron App ?
> Because I fail using it...
> `Cannot GET /Users/raph/Library/Application%20Support/DroopyTempCut/file_1.pdf`
> 
> I can't preview any PDF, with local or distant URL... It seems that the URL isn't good, the console said :
> 
> `GET http://localhost:8080/Users/raph/Library/Application%20Support/DroopyTempCut/file_1.pdf 404 (Not Found)`
> 
> Sorry for my english, but I was so excited about using this Quasar Extension, it's a really important to me to be able to preview a PDF file inside my App...
> 
> Thanks a lot for your work.

Facing the same issue when displaying on electron

**Comment by hawkeye64** [comment]
I am working hard to get a lot of the app-exts converted to Quasar v2. It's on the list, but this one is difficult because of PDFjs being such a pain to work with. I can't say for sure when it'll be done.

**Comment by hawkeye64** [comment]
https://github.com/quasarframework/app-extension-qpdfviewer/issues/87

**Comment by CynthiaBlue** [comment]
I've just installed app-extension-qpdfviewer and am trying it out on my electron/quasar application.  Following the examples, I'm getting a file not found error.  Is my problem the same as this thread... so I cannot use it at this time?  I'm running Quasar version 3.3.3 and Electron version 16.07.  

```

<q-pdfviewer
          type="pdfjs"
          :src="src"
        />
...
 src: '/pdf/pdf-test.pdf'

```



<img width="978" alt="Screen Shot 2022-03-28 at 1 47 24 PM" src="https://user-images.githubusercontent.com/34698014/160475544-3a8ee9b0-8a98-4316-af64-35732f4a6d79.png">

**Comment by hawkeye64** [comment]
Does it work when not running Electron mode?

**Comment by CynthiaBlue** [comment]
> 

I haven't tried it.  My desktop app requires electron.

**Comment by hawkeye64** [comment]
I'll try with Electron in the coming days.
